Description: Disable SVE features in ARM builds that claim NO_ARMV81_OR_HIGHER.
Author: Chris Herborth <chris.herborth@chainguard.dev>
Forwarded: not-needed

diff --git a/contrib/StringZilla-cmake/CMakeLists.txt b/contrib/StringZilla-cmake/CMakeLists.txt
--- a/contrib/StringZilla-cmake/CMakeLists.txt
+++ b/contrib/StringZilla-cmake/CMakeLists.txt
@@ -21,7 +21,8 @@
             "SZ_USE_SVE2_AES=0"
     )
 elseif (ARCH_AARCH64)
-    target_compile_definitions(
+    if (NO_ARMV81_OR_HIGHER)
+        target_compile_definitions(
             _stringzilla PRIVATE
             "SZ_USE_WESTMERE=0"
             "SZ_USE_HASWELL=0"
@@ -29,14 +30,25 @@
             "SZ_USE_ICE=0"
             "SZ_USE_NEON=1"
             "SZ_USE_NEON_AES=1"
+        )
+    else ()
+        target_compile_definitions(
+            _stringzilla PRIVATE
+            "SZ_USE_WESTMERE=0"
+            "SZ_USE_HASWELL=0"
+            "SZ_USE_SKYLAKE=0"
+            "SZ_USE_ICE=0"
+            "SZ_USE_NEON=1"
+            "SZ_USE_NEON_AES=1"
             "SZ_USE_SVE=1"
             "SZ_USE_SVE2=1"
             "SZ_USE_SVE2_AES=1"
-    )
+        )
+    endif ()
 endif ()

 # We need SZ_DYNAMIC_DISPATCH=1 to be public because otherwise the headers will redefine the API (sz_find for example)
 # again and avoid dynamic dispatch.
 target_compile_definitions(_stringzilla PUBLIC "SZ_DYNAMIC_DISPATCH=1")
 target_compile_definitions(_stringzilla PRIVATE "SZ_AVOID_LIBC=0")
-target_compile_definitions(_stringzilla PRIVATE "SZ_DEBUG=0")
\ No newline at end of file
+target_compile_definitions(_stringzilla PRIVATE "SZ_DEBUG=0")
